src/main/java/org/tarantool/TarantoolClientImpl.java

```diff
@@ -599,7 +619,19 @@ public void close() {
         }
     }
 
-    protected boolean isDead(FutureImpl<?> q) {
+    protected class ComposableAsyncOps extends AbstractTarantoolOps<Integer, List<?>, Object, CompletionStage<List<?>>> {
+        @Override
+        public CompletionStage<List<?>> exec(Code code, Object... args) {
+            return (CompletionStage<List<?>>) TarantoolClientImpl.this.doExec(code, args);
+        }
+
+        @Override
+        public void close() {
+            TarantoolClientImpl.this.close();
```

Comment:
It seems other ops classes throws IllegalStateException("You should close TarantoolClient instead.");

Reply:
I've noticed a strange thing: TarantoolClientImpl#close() overides TarantoolClientOps#close and TarantoolClient#close simultaneously. So, ops-instance which was created by  TarantoolClientImpl#syncOps really throws the exception, but calling close() on TarantoolClientImpl#asyncOps leads to call TarantoolClientImpl#close() that closes the client.